Initial commit

a-vishar opened the pull request.

a-vishar committed:
Initial commit

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/Microsoft/MPI.NET?pullRequest=1) <br/>All CLA requirements met.

tminka commented:
This code should not be merged with master until it is compatible with the rest of the codebase.  That means: Build, Config, Deployment, Documentation, Examples, and Tests.

a-vishar committed:
Removed reference to CodeAnalyzer

a-vishar committed:
Added app.config files to allow built.bat to build.

a-vishar committed:
Removed MPI.NET homepage shortcut as it no longer exists.

a-vishar committed:
Added example of Evironment.Run utility method and documentation regarding the method.

a-vishar committed:
Added args to Environment.Run method

a-vishar committed:
Updated documentation & example method for Environment.Run usage

a-vishar committed:
Added tests for Environment disposal via Environment.Run

a-vishar commented:
@tminka 

I have addressed your comments as described below:

- Documentation - Updated to include Environment.Run method.
- Examples - Added an example of how to run the Environment.Run method.
- Build - Added app.config to two existing projects which were missing them, thus preventing the build from functioning correctly.
- Config - No changes required.
- Deployment - No changes required.
- Tests - Added two new tests for Environment.Run method.  It should be noted, that whilst these changes have no impact on the test results, there are some tests which are currently failing such as IntercommunicatorTest which should be investigated in the future.

a-vishar commented:
In relation to the failing test.  It has been documented under [issues](https://github.com/Microsoft/MPI.NET/issues/2) as it is unrelated to this PR:

tminka committed:
Changed order of arguments to Environment.Run, added overloads of Environment.Run. Updated Tutorial and examples to use Environment.Run method.
